packages/discovery-provider/integration_tests/tasks/entity_manager/test_collection_track_relations.py

```diff
@@ -0,0 +1,338 @@
+        for id in [10, 40]:
+            assert not any([relation.track_id == id for relation in relations])
+
+
+# Remove a track from an playlist and then restore it
+restore_removed_track_to_playlist_tx_receipts = {
+    "UpdatePlaylistTracklistUpdate": [
+        {
+            "args": AttributeDict(
+                {
+                    "_entityId": PLAYLIST_ID_OFFSET,
+                    "_entityType": "Playlist",
+                    "_userId": 1,
+                    "_action": "Update",
+                    "_metadata": f'{{"cid": "PlaylistTracklistUpdate", "data": {json.dumps(test_metadata["PlaylistTracklistUpdate"])}, "timestamp": {datetime.timestamp(now)}}}',
+                    "_signer": "user1wallet",
+                }
+            )
+        }
+    ],
+    "RemoveTrackFromPlaylistUpdate": [
+        {
+            "args": AttributeDict(
+                {
+                    "_entityId": PLAYLIST_ID_OFFSET,
+                    "_entityType": "Playlist",
+                    "_userId": 1,
+                    "_action": "Update",
+                    "_metadata": f'{{"cid": "PlaylistTracklistUpdate", "data": {json.dumps(test_metadata["RemoveFromPlaylistTracklistUpdate"])}, "timestamp": {datetime.timestamp(now)}}}',
+                    "_signer": "user1wallet",
+                }
+            )
+        }
+    ],
+    "RestoreTrackToPlaylist": [
+        {
+            "args": AttributeDict(
+                {
+                    "_entityId": PLAYLIST_ID_OFFSET,
+                    "_entityType": "Playlist",
+                    "_userId": 1,
+                    "_action": "Update",
+                    "_metadata": f'{{"cid": "PlaylistTracklistUpdate", "data": {json.dumps(test_metadata["PlaylistTracklistUpdate"])}, "timestamp": {datetime.timestamp(now)}}}',
+                    "_signer": "user1wallet",
+                }
+            )
+        }
+    ],
+}
+
+
+def test_restore_removed_track_to_playlist(app, mocker):
+    db, update_task, entity_manager_txs = setup_db(
+        app, mocker, entities, restore_removed_track_to_playlist_tx_receipts
+    )
+
+    with db.scoped_session() as session:
+        entity_manager_update(
+            update_task,
+            session,
+            entity_manager_txs,
+            block_number=0,
+            block_timestamp=1585336422,
+            block_hash=hex(0),
+        )
+        relations: List[CollectionTrackRelation] = session.query(
+            CollectionTrackRelation
+        ).all()
+        assert len(relations) == 2
+        for id in [20, 30]:
+            assert any(
+                [
+                    relation.track_id == id and not relation.is_delete
+                    for relation in relations
+                ]
+            )
+        for id in [10, 40]:
+            assert not any(
+                [
+                    relation.track_id == id and relation.is_delete
```

Comment:
thinking about what remaining crud actions are technically not covered here, do ya'll think it's worth covering re-order?